### PR TITLE
ROC-4679 Enable admissions e2e - depending on feature toggle

### PR DESCRIPTION
**Pull request by markry-kainos**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-4679

### Change description

Enabling our admissions e2e tests where feature toggle is on.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ X ] No

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -27,9 +27,13 @@ static LinkedHashMap<String, Object> secret(String secretName, String envVar) {
   ]
 }
 
-// Vars needed for AKS testing
+// Vars needed for AKS testing - keep values in sync with: /src/kubernetes/deployment.template.yaml
+// THESE WILL BE OVERWRITTEN IN ASE DEPLOYMENTS WITH TERRAFORM OUTPUTS.TF
+// THESE VARS CAN BE PICKED UP IN INTEGRATION-TEST CODE DIRECTLY
 env.IDAM_URL = 'https://preprod-idamapi.reform.hmcts.net:3511'
 env.CLAIM_STORE_URL = 'http://cmc-claim-store-aat.service.core-compute-aat.internal'
+env.FEATURE_ADMISSIONS = 'true'  // used to selective determine integration tests
```
Comment (markry-kainos) [inline Jenkinsfile_CNP]: @timja ive been battling with this a bit...this is approach im happy with! AKS - env variables used in integration tests defined here, ASE defined in output.tf. Please don't spend any time looking at the actual e2e tests - we are due a major refactoring!!!

File: src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts [hunk 2]
```diff
@@ -15,122 +15,127 @@ const claimantResponseSteps: ClaimantResponseSteps = new ClaimantResponseSteps()
 const checkAndSendPage: ClaimantCheckAndSendPage = new ClaimantCheckAndSendPage()
 const confirmationPage: ClaimantConfirmation = new ClaimantConfirmation()
 
-Feature('Claimant Response').retry(3)
+if (process.env.FEATURE_ADMISSIONS !== undefined && JSON.parse(process.env.FEATURE_ADMISSIONS)) {
```
Comment (markry-kainos) [inline src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts]: @timja this line is of interest too... enabling if toggled

File: infrastructure/output.tf [hunk 3]
```diff
@@ -13,3 +13,27 @@ output "idam_url" {
 output "claim_store_url" {
   value = "${local.claimStoreUrl}"
 }
+
+output "feature_admissions" {
```
Comment (timja) [inline infrastructure/output.tf]: another way that's more full proof i.e. doesn't rely on people adding new ones here is to hit the /info endpoint, it has all feature toggles there

File: src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts (hunk 2 above)
Comment (timja) [reply src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts]: JSON.parse? isn't it just a string?

File: infrastructure/output.tf (hunk 3 above)
Comment (markry-kainos) [reply infrastructure/output.tf]: oh yes!!! yep - much better 👍

File: src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts (hunk 2 above)
Comment (markry-kainos) [reply src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts]: you would think but I had to do this to work locally (docker-compose) and on pipeline...maybe not - i will revisit this before any merge

**Comment by markry-kainos** [comment]
@robertparkinson pls review - i think there is definitely a more codecept.io way to do this - using hooks (`event.all.before`) to detect toggle and set global variable. I couldn’t get it working! We can include this as part of refactoring work on the e2e

**Comment by markry-kainos** [comment]
thanks @singh-surjeet - I added README as agreed: https://github.com/hmcts/cmc-citizen-frontend/pull/1109/files#diff-dce70a764e9461067cab9c77ebc6aa1a - will merge when I get green build 👍

**Pull request merged**